apps/confidential/confidential.go

```diff
@@ -458,6 +459,29 @@ func WithLoginHint(username string) interface {
 	}
 }
 
+// WithDomainHint pre-populates the login prompt with the IdP domain.
```

Comment:
Will there be a situation that the app developer wants to use the default "without domain hint" behavior? If we pre-populate the domain hint with the IdP domain, how does the app developer opt out?

Reply:
this comment was wrong and has been fixed. Should have said 
`// WithDomainHint adds the IdP domain as domain_hint query parameter in the auth url.`

It is not prepopulated and the user has to use the `client.WithDomainHint("contoso.com")` to ask to use domain_hint in the auth url